src/quest_manager/tests/test_models.py

```diff
@@ -381,6 +382,51 @@ def test_submission_get_previous(self):
         second_sub = QuestSubmission.objects.create_submission(user=self.student, quest=repeat_quest)
         self.assertEqual(first_sub, second_sub.get_previous())
 
+    def test_submission_get_previous_automatic_fix_ordinal(self):
+        """Submissions that have the same ordinals will be automatically fixed"""
+        repeat_quest = baker.make(Quest, name="repeatable-quest", max_repeats=-1)
+        first_sub = baker.make(QuestSubmission, user=self.student, quest=repeat_quest, semester=self.semester)
+        self.assertIsNone(first_sub.get_previous())
+        self.assertEqual(first_sub.ordinal, 1)
+        # need to complete so can make another
+        first_sub.mark_completed()
+        SiteConfig.get().set_active_semester(first_sub.semester.id)
+        second_sub = QuestSubmission.objects.create_submission(user=self.student, quest=repeat_quest)
+        self.assertEqual(second_sub.ordinal, 2)
+        second_sub.mark_completed()
+
+        self.assertEqual(first_sub, second_sub.get_previous())
+
+        third_sub = QuestSubmission.objects.create_submission(user=self.student, quest=repeat_quest)
+        self.assertEqual(third_sub.ordinal, 3)
+        third_sub.mark_completed()
+
+        fourth_sub = QuestSubmission.objects.create_submission(user=self.student, quest=repeat_quest)
+        self.assertEqual(fourth_sub.ordinal, 4)
+        fourth_sub.mark_completed()
+
+        # Make the 3rd submission's ordinal same with the second
+        old_third_submission_ordinal = third_sub.ordinal
+        third_sub.ordinal = second_sub.ordinal
+        third_sub.save()
+
```

Comment:
At this point, please check explicitly that  "MultipleObjectsReturned" exception is thrown when you try to `.get(ordinal=second_sub.ordinal)`

I see you are sort of checking that by mocking and verifying that _fix_ordinal is called, but the method could change in the future, and we want to verify why it's called, ie because MultipleObjectsReturned!

Reply:
ooh. that's what you meant. thanks for clarifying